dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -220,10 +220,12 @@ def set_model_parameters(self, *, parameterFilePath=None, **kwargs):
         self.steering_parameters['npath'] = str(parent_dir / Path(cfg.steering_parameters.npath))
         self.steering_parameters['epath'] = str(parent_dir / Path(cfg.steering_parameters.epath))
 
-        # save updated config to yaml file
+        # save updated config to yaml files
         cfg_filename = parent_dir / f'{self._model_identifier}.yaml'
         cfg.to_yaml(cfg_filename)
-        logger.warning(f'The model parameters are saved to {cfg_filename}.')
+        cfg_filename_step = parent_dir / f'{self._model_identifier}_{self.step_count}.yaml'
+        cfg.to_yaml(cfg_filename_step)
+        logger.warning(f'The model parameters are saved to {cfg_filename} and {cfg_filename_step}.')
```

Comment:
It is not clear why the configurations (`cfg`) are stored twice, i.e. `cfg_filename` and `cfg_filename_step`. The function `set_model_parameters` is usually run once at the beginning when `step_count` is 0. Also, tests are not added in `test_model.py::test_set_model_parameters` to verify these changes.

Reply:
Based on our discussion, I will
* remove the part where it stores in `cfg_filename`, only storing in `cfg_filename_step`.
* when continuing from a stored model, find the config file that has the highest step count that is not over the step where the process will continue.
* add tests to test all this functionality

Reply:
I will also see if I can migrate the config parameters into a member of the PovertyTrapModel, as opposed to copying the values of the config.
Copying the values means we cannot just save the model parameters at will, because we might have lost some (non-copied) parameters. Migrating to a Config member of PovertyTrapModel would mean we can save the complete config at will (and I can move saving the config into a separate method).

Reply:
As of commit bfbb990e2d59dc48315eb6c9e6aac69afd9278d7 these features have been implemented, together with some tests.

Reply:
TODO: add 'make unique' functionality for when restarting multiple runs from the same step.